### PR TITLE
Optimisation of apt update

### PR DESCRIPTION
**Pull request by Flashwalker**
Added two functions: update_repo and update_ppa to optimize apt update

### CODE REVIEW

**Comment by philclifford** [comment]
This fixes #428

**Review by natanjunges — APPROVED** [review]

**Review by Flashwalker — COMMENTED** [review]
ok

**Review by natanjunges — CHANGES_REQUESTED** [review]
Please rebase with main.

**Comment by Flashwalker** [comment]
> Please rebase with main.

rebase and push?

**Comment by natanjunges** [comment]
If you go to your branch page in GitHub, you will see it is "3 commits ahead, 10 commits behind" the main branch (or something similar). It needs to get caught up with the main branch (the 10 commits it is behind), so the permission error the test workflow is complaining about gets fixed. If there are no merging conflicts, it should be pretty simple, just click on "Sync fork", and then on the green button that appears. If a red button appears instead, do not click on it, as it would wipe out all of your contributions. In this case, the merge conflicts would have to be resolved locally using the command line.

**Comment by Flashwalker** [comment]
> If you go to your branch page in GitHub, you will see it is "3 commits ahead, 10 commits behind"...

Ok, done.
Now it's just: "4 commits ahead"

File: deb-get [hunk 1]
```diff
@@ -200,10 +200,17 @@ function upgrade_apt() {
     ${ELEVATE} apt-get -q -o Dpkg::Progress-Fancy="1" -y upgrade
 }
```
Comment (philclifford) [inline deb-get:202]: Permissions  got changed  100755 → 100644
please could you change them back to 755

Comment (natanjunges) [reply deb-get:202]: Yes, it seems that whenever GitHub does a rebase from the web interface, it ignores the permissions. I opened #553 to work around those cases in the test workflow.

**Review by natanjunges — APPROVED** [review]

**Review by natanjunges — CHANGES_REQUESTED** [review]
Please rebase with main. As there is already a function named `update_repo`, rename yours (something like `update_apt_repo`).

**Comment by Flashwalker** [comment]
Github suggested me to discard 4 commits. Not sure it was right, but done

**Comment by natanjunges** [comment]
You should not have discarded them, as they were your contributions. Please recommit them and reopen this PR.

**Comment by Fuseteam** [comment]
oh dear, i'm not sure if they know how to do that

**Comment by Flashwalker** [comment]
> oh dear, i'm not sure if they know how to do that

I'd better make a new PR

**Comment by Flashwalker** [comment]
new PR: https://github.com/wimpysworld/deb-get/pull/657

**Pull request closed without merging**